refactor(server): get `PORT` from env var

ramfox opened the pull request:
Also adding more potentially "reservedUsernames" that may conflict with
app routes

b5 commented on lib/actions/dataset.js:
```diff
@@ -284,17 +284,21 @@ function prepFiles (dataset = {}, transformScript = '', vizScript = '', bodyData
+    if (bodyData.constructor === File) {
+      body = bodyData
+    } else if (bodyData.length) {
+      console.log(bodyData, bodyData.length)
```
stray console.log

b5 commented on lib/components/editor/EditBody.js:
```diff
@@ -82,6 +98,67 @@ export default class EditBody extends Base {
+  }
+
+  handleDragEnd (e) {
+    console.log('drag end')
```
stray console.log

b5 commented on lib/components/editor/EditBody.js:
```diff
@@ -82,6 +98,67 @@ export default class EditBody extends Base {
+  handleDragOver (e) {
+    e.stopPropagation()
+    e.preventDefault()
+    console.log('dragover')
```
stray console.log

b5 commented on lib/reducers/editor.js:
```diff
@@ -157,13 +157,19 @@ export default function editorReducer (state = initialState, action) {
-      dataset = {}
-      datasetSections.forEach(s => { dataset[s] = state.dataset[s] })
-      return Object.assign({}, state, { dirty: true, dataset, body })
+      return Object.assign({}, state, { dirty: true, dataset, body, colOrder, rowOrder, bodyView })
```
great integration with existing body state field!

The only thing I'd also like to see is _clearing_ the body file from the state tree on successful save. We could accomplish this just by sending an `EDITOR_INIT_DATASET` action after a successful save. What matters is we shouldn't hang on to a reference to a file that's potentially large for any longer than we have to. Successful save == we done.

We may later want to consider clearing the file when navigating away, but warning the user before letting them leave the editor, all in the name of preserving how much memory we're consuming for the state tree.

ramfox replied:
dope, good call!

ramfox committed:
refactor(server): get `PORT` from env var
Also adding more potentially "reservedUsernames" that may conflict with
app endpoints

Arqu commented on server/index.js:
```diff
@@ -22,7 +22,7 @@ var server = app.listen(PORT, () => {
 // must be updated if routes change, particularly routes with 2 segments as
 // those are the routes at risk for being incorrectly matched
-const reservedUsernames = ['ipfs', 'login', 'workflow'] 
+const reservedUsernames = ['ipfs', 'login', 'workflow', 'admin', 'settings', 'signin', 'signout', 'signup', 'userimage', 'search'] 
```
`oauth`, `identity`, `client`, `issues`, `follow`, `unpublish`

ramfox replied:
ahhh yes. Sorry I don't want you to this I ignored you when you linked to the backend routes, I must have misunderstood. I didn't think that the react app would ever request these routes. added them!

Arqu reviewed (changes requested):
Can we add these too, once ready, I can merge once I settle things in CI

Arqu reviewed (approved):
🚀
